content/how-it-works.md

```diff
@@ -0,0 +1,30 @@
+title: complate Internals and Adaptors
+h1: complate Internals and Adaptors
+subtitle: How it works
+
+At its core, complate is a JavaScript library for generating HTML.
+It is typically used with [JSX](what-is-jsx.html)
+to provide a view layer in an HTTP environment.
+
+JSX itself is not executable,
+so we need a compiler to turn it into executable JavaScript.
+In addition, multiple JSX/JavaScript modules are usually combined into a single "bundle" file,
+as this avoids relying on any particular engine's module-loading mechanism.
+
+![multiple JSX and JavaScript files are transformed and combined into a single bundle](assets/img/how-it-works-1.svg)
+
+The resulting bundle is loaded by the host environment,
+i.e. the respective adaptor
+(depending on the application's primary technology stack).
+It exposes a set of view functions which
+assemble components and constitute a bridge between the host environment and components:
+The host environment invokes these functions via the adaptor while passing parameters,
+which then generates HTML.
+
+![transpile and bundle](assets/img/how-it-works-2.svg)
+
+Depending on the adaptor, parameters might be serialized (JSON-style primitives) or native objects (e.g. database records).
+Ideally complate [writes its HTML directly to the HTTP response stream](faq.html#progressive-rendering),
+though it can also be coerced into returning an HTML string instead.
+
+![request/response cycle](assets/img/how-it-works-3.svg)
```

Comment:
this diagram seems too ... busy? to be easily understood

Reply:
I think its ok for now. We can simplify it later, if we get too many questions about it.

Reply:
It's funny how you assume that anyone will actually read this.

Reply:
I'll be honest. My brain is currently the consistency and quality of overdone oatmeal, so I basically looked over the page and thought: "YES! Finally the last missing piece of the documentation! Let's merge this as soon as possible so I can check it off my checklist"

So the quality of my review is... questionable 🗑 .

Reply:
That's perfectly understandable, so I'll just go ahead and merge it; done > perfect etc.